pkg/dynamicinstrumentation/codegen/c/maps.h

```diff
@@ -10,6 +10,23 @@ struct {
     __uint(max_entries, 1 << 24);
 } events SEC(".maps");
 
+
+// The param_stack map is used as a stack for the location expressions
+// to operate on values and addresses.
+struct inner_param_stack {
+    __uint(type, BPF_MAP_TYPE_STACK);
+    __uint(max_entries, 2048);
+    __uint(value_size, sizeof(__u64));
+};
+
+// The param_stacks map is to set up a unique stack for each CPU.
+struct {
+    __uint(type, BPF_MAP_TYPE_ARRAY_OF_MAPS);
+    __uint(max_entries, 32); // we arbitrarily set this to support 32 CPUs
```

Comment:
I'd suggest setting this to `0` to ensure that userspace is properly setting it.

Reply:
Ah, so if I set to 0 we can override it in userspace?

Reply:
You can always override in userspace, but setting to 0 means it will fail to load if you fail set a value in userspace

Reply:
Yea making this change produces this error (why I originally set it to an arbitrary number):

```
2025-03-24 15:43:54 CDT | SYS-PROBE | INFO | (pkg/dynamicinstrumentation/diconfig/config_manager.go:287 in applyConfigUpdate) | Couldn't load and attach bpf programs could not load bpf collection for probe g504163d-f367-4522-8905-fe8bc34eb975: using replacement map param_stacks: MaxEntries: 4 changed to 0: map spec is incompatible with existing map
```

Reply:
I suppose if we can fully make the outer map in userspace and retrieve it in bpf space then we can set length in Go safely. Is there a way to do that without having a map (with 1 entry) holding this outer map, which holds the stack maps?

Reply:
maps are always created in userspace. The loader does this (simplified):
1. read object file and collect map specs
1. create maps for each spec with the `bpf` syscall, kernel gives back a `fd`
1. modify ebpf bytecode of programs to reference the `fd`
1. load ebpf programs into kernel

By changing the spec in `spec.Maps["param_stacks"]`, it is the same as if it was changed in the object file

Reply:
> `map spec is incompatible with existing map`

this is because `MapReplacements` is trying to ensure `spec.Maps["param_stacks"]` matches what you pass to `MapReplacements["param_stacks"]`. This is so the ebpf code which is expecting a specific map spec continues to work.

You don't need `MapReplacements` here at all for `param_stacks`.

Reply:
If that's the case, why would I be getting the above pasted error?

Reply:
Ah I see then, So I can edit the spec of the outer map before loading. This was because I was making a new map.

Reply:
Yes, exactly.